File: python/12.py
import math,functools

primes = [2]


def divisors():
    _sum = 1
    iterator = 2
    totalDivisors = 0
    while True:
        divisorArr = []
        number = iterator + _sum
        localNumber = number

        for i in primes:
            divCounter = 1
            if i <= (number / 2) + 1:

                while localNumber % i == 0:
                    localNumber=int(localNumber/i)
                    divCounter += 1
                if divCounter>1:    divisorArr.append(divCounter)


        if localNumber > 1:
            dLocalNumber = localNumber
            for i in range(math.floor(localNumber/2)+1,1,-1):
                lDivCounter = 1
                while dLocalNumber%i==0:
                    lDivCounter+=1
                    dLocalNumber=dLocalNumber/i
                if lDivCounter>1:
                    divisorArr.append(lDivCounter)
                    primes.append(i)
            if dLocalNumber > 1:
                primes.append(localNumber)

        if len(divisorArr)>0:
            totalDivisors = functools.reduce(lambda  a,b: a*b, divisorArr)
            if totalDivisors>500:   return number
        _sum += iterator
        iterator+=1


print(divisors())

File: javascript/12.js
let primes=[2];

function divisors(){
    let totalDivisors = 0; 
    let sum=1;
    for (let i = 2; 1; i++) {        
        let divisors =[];
        let number = i+sum;
        let localNumber=number;

        
        primes.forEach(element => {
            let divCounter=1;
            if(element<=number/2){
                while(localNumber%element==0){
                    divCounter++;
                    localNumber=localNumber/element;
                }
                if(divCounter>1)   divisors.push(divCounter);             
            }
                        
        });

        if(localNumber>1){
            let nLocalNumber = localNumber;
            for(let j=Math.floor(localNumber/2)+1;j>1;j--){
                let nDivCounter=1;
                while(nLocalNumber%j==0){
                    nDivCounter++;
                    nLocalNumber=nLocalNumber/j;
                }
                if(nDivCounter>1) {
                    divisors.push(nDivCounter);
                    primes.push(j);
                }
            }
            if(nLocalNumber>1) {
                primes.push(localNumber); 
                divisors.push(2);
            }       
        }        

        sum+=i;
        if(divisors.length>0) totalDivisors = divisors.reduce((a,b) => a*b);        
        if(totalDivisors>500){
            return(number);            
        }
    }
        
    
}

console.log(divisors())